### PR TITLE
GitHub action workflow for root server deployment

### PR DESCRIPTION
**Pull request by kumarnarendra701**

### CODE REVIEW

File: .github/workflows/at_server_dev_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: root_server
+on:
+  push:
+    branches:
+      - trunk
```
Comment (cpswan) [inline .github/workflows/at_server_dev_deploy.yaml:5]: Can we filter on just changes to the at_root code?

File: .github/workflows/at_server_dev_deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+jobs:
+  Docker_Build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (cpswan) [inline .github/workflows/at_server_dev_deploy.yaml]: Should be 3.0.2 (and other actions should have full latest version pinned)

File: .github/workflows/at_server_prod_deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+name: root_server
+on:
+  push:
+    tags:
+      - 'v*.*.*'
```
Comment (cpswan) [inline .github/workflows/at_server_prod_deploy.yaml]: We're going to need to use different tags, as v is used for secondary versions - I'd suggest r

File: .github/workflows/at_server_dev_deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      
+      # Build the Docker image for Dev
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          file: at_root/at_root_server/Dockerfile
+          context: at_root/at_root_server
+          push: true
+          tags: atsigncompany/root:dev_env-${{ env.BRANCH }}-gha${{ github.run_number }}
```
Comment (cpswan) [inline .github/workflows/at_server_dev_deploy.yaml]: can we also have a :dev-env tag that acts as the latest

File: .github/workflows/at_server_prod_deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      
+      # Build the Docker image for Dev
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          file: at_root/at_root_server/Dockerfile
+          context: at_root/at_root_server
+          push: true
+          tags: atsigncompany/root:prod-${{ env.BRANCH }}-gha${{ github.run_number }}
```
Comment (cpswan) [inline .github/workflows/at_server_prod_deploy.yaml]: there should also be a :prod tag for the latest prod release

**Review by cpswan — CHANGES_REQUESTED** [review]
I was expecting these workflows to go into at_server.yaml, though I guess they can run standalone.

File: .github/workflows/at_server_dev_deploy.yaml (hunk 2 above)
Comment (kumarnarendra701) [reply .github/workflows/at_server_dev_deploy.yaml]: Fixed

File: .github/workflows/at_server_prod_deploy.yaml (hunk 3 above)
Comment (kumarnarendra701) [reply .github/workflows/at_server_prod_deploy.yaml]: Fixed

File: .github/workflows/at_server_prod_deploy.yaml (hunk 5 above)
Comment (kumarnarendra701) [reply .github/workflows/at_server_prod_deploy.yaml]: Fixed

File: .github/workflows/at_server_dev_deploy.yaml (hunk 4 above)
Comment (kumarnarendra701) [reply .github/workflows/at_server_dev_deploy.yaml]: fixed

File: .github/workflows/at_server_dev_deploy.yaml (hunk 1 above)
Comment (kumarnarendra701) [reply .github/workflows/at_server_dev_deploy.yaml:5]: I have changed it to at_root.

Comment (cpswan) [reply .github/workflows/at_server_dev_deploy.yaml:5]: I don't think changing the `name:` achieves what I'd like to see here. The `on:` parameter needs to be much more selective.

**Review by cpswan — CHANGES_REQUESTED** [review]

File: .github/workflows/at_server_dev_deploy.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@ name: at_root
 on:
   push:
     branches:
-      - trunk
+      - ci-cd-test
```
Comment (cpswan) [inline .github/workflows/at_server_dev_deploy.yaml]: I think trunk still needs to be the branch we use, as `ci-cd-test` will go once this PR is merged.

**Review by cpswan — CHANGES_REQUESTED** [review]

File: .github/workflows/at_server_dev_deploy.yaml (hunk 6 above)
Comment (kumarnarendra701) [reply .github/workflows/at_server_dev_deploy.yaml]: This is not a final request for review, just added the ci-cd-test branch to test it.

File: .github/workflows/at_server_dev_deploy.yaml (hunk 2 above)
Comment (cpswan) [reply .github/workflows/at_server_dev_deploy.yaml]: I'm not seeing that fix

Comment (kumarnarendra701) [reply .github/workflows/at_server_dev_deploy.yaml]: Earlier added to deploy workflow only, Now fixed for all YAML files.

**Review by cpswan — APPROVED** [review]
There are a few files here with no changes of substance, but I don't see any harm.

**Pull request merged**